materialize-boilerplate/transactions_stream.go

```diff
@@ -199,6 +206,19 @@ func (l *transactionsStream) maybeDelayAcknowledgement() error {
 				return l.ctx.Err()
 			case <-time.After(d):
 			}
+
+			// We trigger dbt job only if the delay is taking effect, to avoid bursting
+			// dbt job triggers during backfills
```

Comment:
One question worth posing here: have cases where a materialization has so much data that it always skips the default delay? In those cases it might seem like dbt job trigger is not working at all, but it is being always postponed... 🤔

Reply:
There are definitely active tasks where a materialization has some much continuous data that when combined with the backpressure of the destination system, the transactions are always greater than 1MM documents and the delay would always be skipped.

Fortunately these are also cases where the tasks are configured with an explicit delay of `0s`, so it doesn't really matter. I doubt this will always be the case though. Our current delay triggering mechanism based on the number of documents in a transaction is fundamentally flawed but it is the best we can do right now.